### PR TITLE
Xw 417 | pass staging to MWRequest

### PR DESCRIPTION
**Pull request by bognix**
https://wikia-inc.atlassian.net/browse/XW-417

### CODE REVIEW

File: server/facets/editorPreview.ts [hunk 1]
```diff
@@ -13,7 +13,9 @@ function editorPreview (request: Hapi.Request, reply: Hapi.Response): void {
 	var wikiDomain: string = Utils.getCachedWikiDomainName(localSettings, request),
 		parserOutput: string = request.payload.parserOutput,
 		mwHash: string = request.payload.mwHash,
-		article = new Article.ArticleRequestHelper({wikiDomain: wikiDomain});
+		//@TODO remove when https://github.com/Wikia/chef-repo/pull/6681 fixed
```
Comment (bkoval) [inline server/facets/editorPreview.ts]: Do we have a tracking ticket for that? If not, should we have it? ;)

File: server/lib/MediaWiki.ts [hunk 2]
```diff
@@ -31,9 +33,18 @@ class BaseRequest {
...
 		this.wikiDomain = params.wikiDomain;
 		this.headers = params.headers;
+		this.staging = params.staging;
 	}
 
 	fetch(url: string): any {
+		//@TODO remove when https://github.com/Wikia/chef-repo/pull/6681 fixed
+		if (this.staging) {
+			if (url.indexOf('?') > -1) {
+				url = url + "&staging=" + this.staging
```
Comment (Warkot) [inline server/lib/MediaWiki.ts]: To reduce repetition in the code you could write something like this:
```
connector = url.indexOf('?') >= 0 ? '&' : '?';
url += connector + "staging=" + this.staging;
```
And don't forget about semicolons `;`. :cop:

Comment (rogatty) [inline server/lib/MediaWiki.ts]: It's a temporary solution for this weekend. Let's not waste time :)

File: server/facets/editorPreview.ts (hunk 1 above)
Comment (bognix) [inline server/facets/editorPreview.ts]: yes: https://wikia-inc.atlassian.net/browse/XW-438

File: server/facets/showMainPageSection.ts [hunk 3]
```diff
@@ -8,9 +8,12 @@ import processCuratedContentData = require('./operations/processCuratedContentDa
 
 function showSection (request: Hapi.Request, reply: Hapi.Response): void {
 	var wikiDomain: string = Utils.getCachedWikiDomainName(localSettings, request),
+		//@TODO remove when https://github.com/Wikia/chef-repo/pull/6681 fixed
+		staging = request.headers['x-staging'],
 		params: MainPageRequestParams = {
 			sectionName: decodeURIComponent(request.params.sectionName) || null,
-			wikiDomain: wikiDomain
+			wikiDomain: wikiDomain,
+			staging: staging
```
Comment (Warkot) [inline server/facets/showMainPageSection.ts]: You could drop creation of the `staging` variable and just write `staging: request.headers['x-staging']`.

File: server/lib/MediaWiki.ts (hunk 2 above)
Comment (Warkot) [inline server/lib/MediaWiki.ts]: kk :shipit:

**Comment by rogatty** [comment]
Tests are passing locally, there are some timeouts on Jenkins.

**Pull request merged**